### PR TITLE
Pagination feature

### PR DESCRIPTION
**Pull request by JKlarlund**
- Only displaying 32 cheeps per page for user & page queries
- Incorrect query defaults to first page

### CODE REVIEW

File: src/Chirp.Razor/Pages/Public.cshtml.cs [hunk 1]
```diff
@@ -8,16 +8,27 @@ namespace Chirp.Razor.Pages;
...
     {
         _service = service;
+        Cheeps = _service.GetCheeps();
+
     }
 
     public ActionResult OnGet()
     {
-        Cheeps = _service.GetCheeps();
+        PageNumber = int.TryParse(QueryPage, out var page) && 0<=page && page<=MathUtil.PageAmount(Cheeps) ? page : 1;
```
Comment (alexop1000) [inline src/Chirp.Razor/Pages/Public.cshtml.cs:26]: Could use some formatting in this section, probably other places too

File: src/Chirp.Razor/Pages/UserTimeline.cshtml.cs [hunk 2]
```diff
@@ -18,6 +24,12 @@ public UserTimelineModel(ICheepService service)
     public ActionResult OnGet(string author)
     {
         Cheeps = _service.GetCheepsFromAuthor(author);
+        PageNumber = int.TryParse(QueryPage, out var page) && 0 <= page && page <= MathUtil.PageAmount(Cheeps) ? page : 1;
```
Comment (alexop1000) [inline src/Chirp.Razor/Pages/UserTimeline.cshtml.cs:27]: Instead of repeating the code, maybe make a method to do this?

**Review by alexop1000 — CHANGES_REQUESTED** [review]

File: src/Chirp.Razor/Pages/UserTimeline.cshtml.cs (hunk 2 above)
Comment (JKlarlund) [reply src/Chirp.Razor/Pages/UserTimeline.cshtml.cs:27]: Great idea, I will do that!

File: src/Chirp.Razor/Pages/Public.cshtml.cs (hunk 1 above)
Comment (JKlarlund) [reply src/Chirp.Razor/Pages/Public.cshtml.cs:26]: > Could use some formatting in this section, probably other places too

I will look into that 👍

Comment (JKlarlund) [reply src/Chirp.Razor/Pages/Public.cshtml.cs]: Maybe #68f8c58f29f9a3dcae5ca433ecbc23c2661fd4e9 resolves this?

Comment (JKlarlund) [reply src/Chirp.Razor/Pages/Public.cshtml.cs]: Maybe 68f8c58f29f9a3dcae5ca433ecbc23c2661fd4e9 solves it?

File: src/Chirp.Razor/Pages/Public.cshtml [hunk 3]
```diff
@@ -1,18 +1,19 @@
-﻿@page "/"
+﻿@page "/page"
```
Comment (MrBStones) [inline src/Chirp.Razor/Pages/Public.cshtml:1]: This should probably still be the front page.

File: src/Chirp.Razor/Pages/UserTimeline.cshtml [hunk 4]
```diff
@@ -30,4 +32,5 @@
     {
         <em>There are no cheeps so far.</em>
     }
+    <small>page @Model.PageNumber</small>
```
Comment (MrBStones) [inline src/Chirp.Razor/Pages/UserTimeline.cshtml:35]: add buttons here to change pages

File: src/Chirp.Razor/MathUtil.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+using Chirp.Core.Classes;
+
+namespace Chirp.Razor;
+
+public class MathUtil
```
Comment (MrBStones) [inline src/Chirp.Razor/MathUtil.cs:5]: does this have to be its own class?

File: src/Chirp.Razor/Pages/Public.cshtml [hunk 6]
```diff
@@ -27,6 +28,7 @@
     }
     else
     {
-        <em>There are no cheeps so far.</em>
+    <em>There are no cheeps so far.</em>
     }
-</div>
+    <small>page @Model.PageNumber</small>
```
Comment (MrBStones) [inline src/Chirp.Razor/Pages/Public.cshtml:33]: add buttons here too

Comment (MrBStones) [reply src/Chirp.Razor/Pages/Public.cshtml:33]: Probably would be a good idea make a function or something for this

**Review by MrBStones — CHANGES_REQUESTED** [review]

File: src/Chirp.Razor/Pages/Public.cshtml (hunk 3 above)
Comment (JKlarlund) [reply src/Chirp.Razor/Pages/Public.cshtml]: Yeah for sure! Fixed 71a789b5b6b3b00935442835cd89b51b553a1ab4

File: src/Chirp.Razor/MathUtil.cs (hunk 5 above)
Comment (JKlarlund) [reply src/Chirp.Razor/MathUtil.cs]: No good point 🤣. See db3a9a0e6027294c887fa01a4adccfba676786ba

File: src/Chirp.Razor/Pages/UserTimeline.cshtml (hunk 4 above)
Comment (JKlarlund) [reply src/Chirp.Razor/Pages/UserTimeline.cshtml]: Does 64381b11297597d1d2ec9577780e7b80e1754170 solve this?

File: src/Chirp.Razor/Pages/Public.cshtml (hunk 6 above)
Comment (JKlarlund) [reply src/Chirp.Razor/Pages/Public.cshtml]: Thanks for very good feedback!
64381b11297597d1d2ec9577780e7b80e1754170 hopefully fixes 

> add buttons here too

What should I make a function for exactly?

File: src/Chirp.Razor/Pages/UserTimeline.cshtml (hunk 4 above)
Comment (MrBStones) [reply src/Chirp.Razor/Pages/UserTimeline.cshtml]: Yep ☺️

File: src/Chirp.Razor/Pages/Public.cshtml (hunk 6 above)
Comment (MrBStones) [reply src/Chirp.Razor/Pages/Public.cshtml]: I only meant to make it so we can reuse the code for buttons but this is good 👍

**Review by MrBStones — CHANGES_REQUESTED** [review]
I tried adding some extra cheeps to the mock DB and for some reason it seems like the pages don't switch? The buttons change the page correctly in the URL but the content does'nt change.

**Review by MrBStones — APPROVED** [review]
okay it works now

**Comment by MrBStones** [comment]
Now you just need to merge main into this branch and fix the merge errors before merging the pull request

**Comment by JKlarlund** [comment]
Should be all good with this branch, otherwise lmk <3

**Pull request merged**